### DISCUSSION TITLE
Modal disable click outside

### DISCUSSION THREAD

**Discussion by jhegele** [mantinedev/mantine#371] [Q&A] [answered]
It doesn't look like there's an out-of-the-box way to disable the click outside functionality of a modal. I am building a form in a modal and it would be great to have the ability to disable click outside to avoid accidentally losing the form. I tried the useClickOutside hook but the Modal component doesn't accept a ref so that's a non-starter. Any thoughts on how I can accomplish this without rolling my own custom modal component?

Thanks!

**Comment by rtivital** [comment, marked as answer]
Hi @jhegele, I've released a patch that adds this feature, please update to `3.1.3` and disable click outside with:

```tsx
<Modal closeOnClickOutside={false}>
  Your form
</Modal>
```

**Comment by jhegele** [comment]
Thanks for the quick response @rtivital! Works perfectly.

**Comment by JosefHartsough** [comment]
Is it possible to change the behavior for when text is selected inside the modal? For example if you highlight text and move the mouse outside of the modal it will close. This doesn't seem like the intended behavior.

**Reply by UltraInstinct05** [reply]
Maybe you could create a sandbox that replicates the issue ? I can't reproduce this. If I select some text and move the mouse outside (not click outside), it doesn't close automatically.

**Reply by JosefHartsough** [reply]
I apologize, I should have been more specific. Once the mouse is let go, the modal will close.

**Reply by JosefHartsough** [reply]
It seems like it should only close if you click outside of the modal, not if you highlight text and let go of the mouse outside of the modal.

**Reply by UltraInstinct05** [reply]
Doesn't `closeOnClickOutside={false}` work for your case ?

**Reply by rtivital** [reply]
@Ultra-Instinct-05 I got the issue, it will be fixed in next minor
